website/content/docs/operations/aws-oidc-provider.mdx

```diff
@@ -0,0 +1,361 @@
+to create an appropriate IAM role for federated users. This will be specific to your use case.
+The following example sets S3 bucket access.
+
+```hcl
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Federated"
+      identifiers = ["arn:aws:iam::<AWS_ACCOUNT_ID>:oidc-provider/$OIDC_PROVIDER"]
+    }
+
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    condition {
+        test = "StringEquals"
+        variable = "$OIDC_PROVIDER:aud"
+        values = "aws"
+    }
+  }
+}
+
+resource "aws_iam_role" "s3_all_access_role" {
+  name = "s3_all_access_role"
+
+  # Terraform's "jsonencode" function converts a
+  # Terraform expression result to valid JSON syntax.
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+            "s3:*",
+            "s3-object-lambda:*"
+        ]
+        Resource = "*"
+      },
+    ]
+  })
+
+  tags = {
+    tag-key = "tag-value"
+  }
+}
+
+resource "aws_iam_policy" "policy" {
+  name        = "nomad-oidc-policy"
+  description = "A policy for federated Nomad OIDC"
+  policy      = data.aws_iam_policy_document.assume_role.json
+}
+
+resource "aws_iam_role_policy_attachment" "test-attach" {
+  role       = aws_iam_role.s3_all_access_role.name
+  policy_arn = aws_iam_policy.policy.arn
+}
+```
+
+This configuration requires the following information:
+- `<AWS_ACCOUNT_ID>`: The ID of the AWS account where the IAM role from the previous step was
+created.
+
+## Update the Nomad server configuration
+
+After you configure AWS, modify the Nomad server configuration file. Add the
+[`oidc_issuer` attribute](/nomad/docs/configuration/server#oidc_issuer) and set the value
+to the domain name for the Nomad cluster. This [enables the HTTP endpoint in Nomad](https://developer.hashicorp.com/nomad/docs/configuration/server#oidc_issuer) that allows third parties to discover Nomad's OIDC
+configuration.
+
+```hcl
+server {
+  enabled = true
+  [...]
+  oidc_issuer = "https://<DOMAIN_NAME>"
+  [...]
+}
+```
+
+This configuration requires the following information:
+- `<DOMAIN_NAME>`: The domain name of your Nomad cluster without the protocol or port.
+
+Restart the Nomad server agent to apply the configuration changes.
+
+## Create and run a sample jobspec file
+
+Create and run a jobspec file to validate your configuration. The following file is named `s3-upload.nomad.hcl`, add the following configuration to it, and
+save the file.
+
+```hcl
+job "s3" {
+  type = "batch"
+  group "bucket" {
+    task "copy" {
+      driver = "docker"
+      config {
+        image = "public.ecr.aws/aws-cli/aws-cli"
+        command = "s3"
+        args = ["cp", "/local/test.txt", "s3://<S3_BUCKET_NAME>/test-nomad.txt"]
+      }
+
+      identity {
+        name = "aws"
+        aud = ["aws"] 
+        file = true
+        env = true
```

Comment:
We probably don't need `env = true`, as we're setting a specific environment variable `AWS_WEB_IDENTITY_TOKEN_FILE` from the `file = true` instead.